python3-sys/src/initconfig.rs

```diff
@@ -0,0 +1,143 @@
+// This entire module is Python 3.8+ and !Py_LIMITED_API only.
+
+use libc::{c_char, c_int, wchar_t, c_ulong};
+use crate::pyport::Py_ssize_t;
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub enum PyStatusType {
+    _PyStatus_TYPE_OK = 0,
+    _PyStatus_TYPE_ERROR = 1,
+    _PyStatus_TYPE_EXIT = 2,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct PyStatus {
+    _type: PyStatusType,
+    func: *const c_char,
+    err_msg: *const c_char,
+    exitcode: c_int,
+}
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
+    pub fn PyStatus_Ok() -> PyStatus;
+    pub fn PyStatus_Error(err_msg: *const c_char) -> PyStatus;
+    pub fn PyStatus_NoMemory() -> PyStatus;
+    pub fn PyStatus_Exit(exitcode: c_int) -> PyStatus;
+
+    pub fn PyStatus_IsError(err: PyStatus) -> c_int;
+    pub fn PyStatus_IsExit(err: PyStatus) -> c_int;
+    pub fn PyStatus_Exception(err: PyStatus) -> c_int;
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct PyWideStringList {
+    length: Py_ssize_t,
+    items: *mut *mut wchar_t,
+}
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
+    pub fn PyWideStringList_Append(list: *mut PyWideStringList, item: *const wchar_t) -> PyStatus;
+    pub fn PyWideStringList_Insert(list: *mut PyWideStringList, index: Py_ssize_t, item: *const wchar_t) -> PyStatus;
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct PyPreConfig {
```

Comment:
(detail) this `PyPreConfig` and `PyConfig` seem to be used only through pointers. Do they really need to be `Copy`, then?

Reply:
Nope. I'm not even sure why I added `Copy`. Maybe I was basing it on patterns I saw elsewhere? I generally like avoiding `Copy` unless absolutely necessary because the implicit nature of `Copy` over `.clone()` can get you in trouble. I'll remove `Copy`.